### PR TITLE
Migrate to gradle (fixes #32)

### LINKED ISSUE #32: Migrate to gradle

### PR DESCRIPTION
**Pull request by baev**

### CODE REVIEW

**Comment by baev** [comment]
retest this please

**Comment by baev** [comment]
retest this please

**Comment by baev** [comment]
retest this please

File: allure-commandline/build.gradle [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    compile project(':allure-generator')
+    compile('com.github.rvesse:airline:2.2.0')
+    compile('com.squareup.retrofit:retrofit:1.9.0')
+    compile('org.apache.commons:commons-exec:1.3')
+    compile('org.eclipse.jetty:jetty-server:9.2.2.v20140723')
+    compile("org.slf4j:slf4j-api:${slf4jVersion}")
+    compile("org.slf4j:slf4j-log4j12:${slf4jVersion}")
+
+    testCompile('junit:junit:4.12')
```
Comment (lanwen) [inline allure-commandline/build.gradle]: use junit5M3 - its cool!

File: allure-generator/src/main/java/org/allurefw/report/plugins/DefaultPluginLoader.java [hunk 2]
```diff
@@ -75,18 +78,32 @@ public DefaultPluginLoader(Path pluginsDirectory) {
...
+        Stream<Path> libs = Files.isDirectory(lib) ? getChildren(lib) : Stream.empty();
+        Stream<Path> pluginJar = Stream.of(pluginDirectory.resolve(PLUGIN_JAR_ENTRY_NAME));
+        return Stream.concat(pluginJar, libs)
+                .map(this::toUrl)
+                .toArray(URL[]::new);
+    }
+
+    private Stream<Path> getChildren(Path directory) throws IOException {
+        try (Stream<Path> children = Files.walk(directory, 1, FileVisitOption.FOLLOW_LINKS)) {
+            //we need to load lazy directory stream to memery and close it
```
Comment (lanwen) [inline allure-generator/src/main/java/org/allurefw/report/plugins/DefaultPluginLoader.java]: memery :)

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        jaxb('net.java.dev.jaxb2-commons:jaxb-fluent-api:2.1.8')
+        jaxb('org.slf4j:slf4j-simple:1.7.21')
+    }
+
+    tasks.withType(Tar){
+        compression = Compression.GZIP
+    }
+
+    repositories {
+        mavenLocal()
```
Comment (lanwen) [inline build.gradle]: its better to use jCenter

Comment (baev) [inline build.gradle]: why?

File: allure-commandline/build.gradle (hunk 1 above)
Comment (baev) [inline allure-commandline/build.gradle]: the PR is not about dependencies upgrade

File: build.gradle (hunk 3 above)
Comment (lanwen) [inline build.gradle]: faster (more mirrors), more secured, bigger repo (it's replacement for central with sync), greater availability than central. All who use gradle usually use jcenter

**Pull request merged**